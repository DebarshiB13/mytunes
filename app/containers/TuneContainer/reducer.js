/*
 *
 * TuneContainer reducer
 *
 */
import { translate } from '@app/components/IntlGlobalProvider/index';
import produce from 'immer';
import get from 'lodash/get';
import { createActions } from 'reduxsauce';

export const initialState = { searchTerm: null, songsData: [], songsError: null };

export const { Types: tuneContainerTypes, Creators: tuneContainerCreators } = createActions({
  requestGetItuneSongs: ['searchTerm'],
  successGetItuneSongs: ['data'],
  failureGetItuneSongs: ['error'],
  clearItuneSongs: []
});

/* eslint-disable default-case, no-param-reassign */
export const tuneContainerReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case tuneContainerTypes.REQUEST_GET_ITUNE_SONGS:
        draft.searchTerm = action.searchTerm;
        break;
      case tuneContainerTypes.SUCCESS_GET_ITUNE_SONGS:
        draft.songsData = action.data;
        draft.songsError = null;
        break;
      case tuneContainerTypes.FAILURE_GET_ITUNE_SONGS:
        draft.songsError = get(action.error, 'message', translate('something_went_wrong'));
        draft.songsData = [];
        break;
      case tuneContainerTypes.CLEAR_ITUNE_SONGS:
        return initialState;
    }
  });

export default tuneContainerReducer;
