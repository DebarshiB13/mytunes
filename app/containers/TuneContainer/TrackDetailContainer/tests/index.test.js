/**
 *
 * Tests for TrackDetailContainer
 *
 */

import React from 'react';
import { renderWithIntl, timeout } from '@utils/testUtils';
import { mapDispatchToProps, TrackDetailContainerTest as TrackDetailContainer } from '../index';
import { tuneContainerTypes } from '../../reducer';

describe('<TrackDetailContainer />', () => {
  let submitSpy;
  let trackDetails;
  beforeEach(() => {
    trackDetails = {
      artistName: 'Sia',
      collectionName: 'abc',
      cardImg: 'https://abc.jpg',
      previewUrl: 'https://bcw.mp3',
      trackId: '122'
    };
    submitSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <TrackDetailContainer trackDetails={trackDetails} dispatchGetTrackDetails={submitSpy} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render Errorcard when trackError is true ', () => {
    const error = 'something_went_wrong';

    const { getByTestId } = renderWithIntl(
      <TrackDetailContainer dispatchGetTrackDetails={submitSpy} trackError={error} />
    );
    expect(getByTestId('track-detail-error')).toBeInTheDocument();
  });

  it('should render TuneCard when trackDetails is not empty', () => {
    const { getByTestId } = renderWithIntl(
      <TrackDetailContainer dispatchGetTrackDetails={submitSpy} trackDetails={trackDetails} />
    );
    expect(getByTestId('tune-card')).toBeInTheDocument();
  });

  it('should dispatch FETCH_TRACK_DETAILS on mount', async () => {
    renderWithIntl(<TrackDetailContainer dispatchGetTrackDetails={submitSpy} />);

    await timeout(500);
    expect(submitSpy).toBeCalled();
  });

  it('should match mapDispatchToProps actions', async () => {
    const songId = '345776';
    const dispatchSpy = jest.fn((fn) => fn);
    const dispatchGetTrackDetailsSpy = jest.fn(() => ({ type: tuneContainerTypes.REQUEST_GET_TRACK_DETAILS, songId }));

    const props = mapDispatchToProps(dispatchSpy);

    const actions = {
      dispatchGetTrackDetailsSpy
    };

    props.dispatchGetTrackDetails(songId);
    expect(dispatchSpy).toHaveBeenCalledWith(actions.dispatchGetTrackDetailsSpy());
  });
});
