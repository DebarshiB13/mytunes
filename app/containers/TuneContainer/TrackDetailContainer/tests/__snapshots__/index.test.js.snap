// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetailContainer /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetailContainer__Container-sc-1jvorpq-0 kZtAhA"
    >
      <div
        class="ant-card ant-card-bordered TrackDetailContainer__CustomCard-sc-1jvorpq-1 iOzOIi"
        padding="1.4"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="TrackDetailContainer__PlaceHolderDiv-sc-1jvorpq-5 OTynr"
          />
          <div
            class="ant-row TrackDetailContainer__StyledRow-sc-1jvorpq-4 dTDiEL"
            style="margin: -12px -8px -12px -8px;"
          >
            <div
              class="ant-col"
              style="padding: 12px 8px 12px 8px;"
            >
              <div
                class="ant-card ant-card-bordered TrackDetailContainer__CustomCard-sc-1jvorpq-1 KEEre"
                maxwidth="30"
              >
                <div
                  class="ant-card-body"
                >
                  <h1
                    class="ant-typography TrackDetailContainer__StyledTitle-sc-1jvorpq-2 cYSwbi"
                    font-size="1.4"
                  >
                    Collection: 
                    abc
                  </h1>
                  <h1
                    class="ant-typography TrackDetailContainer__StyledTitle-sc-1jvorpq-2 dhGsbg"
                    font-size="1.2"
                  >
                    Artist: 
                    Sia
                  </h1>
                  <div
                    class="ant-card-meta TrackDetailContainer__StyledMeta-sc-1jvorpq-3 eHyqvw"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        Country: undefined
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-card-meta TrackDetailContainer__StyledMeta-sc-1jvorpq-3 eHyqvw"
                  >
                    <div
                      class="ant-card-meta-detail"
                    >
                      <div
                        class="ant-card-meta-title"
                      >
                        Genre name: undefined
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col"
              style="padding: 12px 8px 12px 8px;"
            >
              <div
                class="ant-row TrackDetailContainer__StyledRow-sc-1jvorpq-4 dTDiEL"
              >
                <div
                  class="ant-card ant-card-bordered TuneCard__ItemCard-sc-1cl6pnu-0 hjgfUg"
                  data-testid="tune-card"
                  maxwidth="240"
                >
                  <div
                    class="ant-card-cover"
                  >
                    <img
                      alt="example"
                    />
                  </div>
                  <div
                    class="ant-card-body"
                  >
                    <div
                      class="ant-row"
                    >
                      <div
                        class="ant-col"
                      >
                        <h1
                          class="ant-typography TuneCard__StyledT-sc-1cl6pnu-2 eJcBSi"
                          data-testid="artist-name"
                          font-size="1.6"
                        >
                          
                        </h1>
                      </div>
                    </div>
                    <div
                      class="ant-row"
                    >
                      <div
                        class="ant-col"
                      >
                        <div
                          class="ant-card-meta"
                          font-size="1.2"
                        />
                      </div>
                    </div>
                    <div
                      class="ant-row"
                    >
                      <div
                        class="ant-col"
                      >
                        <button
                          class="ant-btn ant-btn-text ant-btn-circle ant-btn-icon-only TuneCard__IconButton-sc-1cl6pnu-1 eahDeJ"
                          data-testid="play-pause-btn"
                          type="button"
                        >
                          <span
                            aria-label="play-circle"
                            class="anticon anticon-play-circle"
                            role="img"
                            style="font-size: 32px;"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="play-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                    <audio
                      data-testid="audio"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
